Add a key-only rpmdb iterator to optimize rpmtsCheck()

pmatilai opened the pull request:
rpmtsCheck() builds some hashes over rpmdb keys, whose buildup is barely measurable in normal install/update/erase transactions, but when repeated thousands of times during rpm -Va, things start adding up. Add a new interface to avoid fetching tonnes of unused data and use it. More details in the commits.

mlschroe commented:
Oooh, did you notice the #if 0 code in ndb/glue.c? I once started the same thing (but used DBC_KEY_SEARCH to select it), but didn't finish the implementation.

mlschroe commented:
You may want to rename `skipdata` to `ii_skipdata` so that it fits in better with the other elements.

pmatilai commented:
Yeah I noticed the #if 0 -block and its similarity, I just assumed it was a leftover/work-in-progress thing related to, well prefix search :)

I remember a fleeting thought "why is there no flags field in the index iterator struct where I could stuff this in", but somehow the thought of putting the flag in the cursor never occurred to me. Maybe thoughts on the long weekend already...

Anyway, I'll rename to ii_skipdata and merge with the ndb #if 0 part now that I know what it is.
We can always switch the implementation to use a cursor flag later if it becomes useful (where it'd really make a difference of course is the pkgdb, but don't have an immediate use-case for that).

pmatilai commented:
Updated to reuse the #if 0'ed codeblock, renamed to ii_skipdata and commit messages adjusted, they were somewhat mixed up with the latter commit speaking about updating backends, which was obviously done in the first commit.

pmatilai commented:
Oh and thanks for the review!
